lldb/source/Plugins/LanguageRuntime/ObjC/AppleObjCRuntime/AppleObjCRuntimeV2.cpp

```diff
@@ -3278,7 +3278,7 @@ bool AppleObjCRuntimeV2::NonPointerISACache::EvaluateNonPointerISA(
       }
 
       // If the index is still out of range then this isn't a pointer.
-      if (index > m_indexed_isa_cache.size())
+      if (index >= m_indexed_isa_cache.size())
```

Comment:
This fixes this bound check, but there's the exact same bounds check above on line 3228. I don't see that mentioned in the blog post's analysis. From the comments above line 3228, that should be changed as well, no?

Reply:
That sounds correct. Let me think a bit about your words